src/v/cloud_topics/reconciler/reconciler.cc

```diff
@@ -224,12 +226,40 @@ ss::future<> reconciler::commit_object(const object_range_info& range) {
      * TODO: we aren't actually replicating an overlay batch here yet. This will
      * come at the time of integration with the data layout STM.
      */
-    range.partition->lro = range.info.last_offset + model::offset(1);
+    const auto& part = range.partition->partition;
+    object_id id = object_id{uuid_t::create()};
+    dl_overlay_object overlay_obj(
+      id,
+      first_byte_offset_t{range.physical_offset_start},
+      byte_range_size_t{
+        range.physical_offset_end - range.physical_offset_start},
+      dl_stm_object_ownership::shared);
```

Comment:
I was ready to comment that this looks like unique object because we replicate it only to a single partition... but then I noticed that this method is called for multiple partitions with the same object id 😵‍💫 

This is surprising here because from this place we can't tell whether the object is unique or shared. And it might as well be unique, i.e. large partition might as well have hit alone the object size limit and take up the whole object.

The ownership should be a parameter! 🎉 It can be stored in `reconciler::object` and then used in commit_object calls ;)